### PR TITLE
hand resource down to drive controller

### PR DESCRIPTION
**Pull request by ryanfkeepers**
hands the backup resource into the drive collection for the handler to use to record as the siteID

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :bug: Bugfix

#### Issue(s)

* #3988

#### Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/internal/m365/collection/drive/handler_utils.go [hunk 1]
```diff
@@ -99,9 +88,9 @@ func augmentItemInfo(
 			Modified:   ptr.Val(item.GetLastModifiedDateTime()),
 			Owner:      creatorEmail,
 			ParentPath: pps,
-			SiteID:     siteID,
+			SiteID:     resource.ID(),
```
Comment (meain) [inline src/internal/m365/collection/drive/handler_utils.go:91]: Ash was bringing up a probability of this causing issues in Ark. I'm not sure how this gets consumed in Ark, but would be useful to check.

**Review by meain — CHANGES_REQUESTED** [review]

File: src/internal/m365/collection/drive/handler_utils.go [hunk 2]
```diff
@@ -99,9 +88,9 @@ func augmentItemInfo(
...
 			Size:       size,
-			WebURL:     weburl,
+			WebURL:     resource.Name(),
```
[the remark this answers is not in the document]
Comment (meain) [reply src/internal/m365/collection/drive/handler_utils.go:93]: Here is where you'll have to change it.

```diff
diff --git a/src/internal/m365/service/groups/backup.go b/src/internal/m365/service/groups/backup.go
index 25210ade3..0bb9a9b44 100644
--- a/src/internal/m365/service/groups/backup.go
+++ b/src/internal/m365/service/groups/backup.go
@@ -93,7 +93,7 @@ func ProduceBackupCollections(
 			}
 
 			for _, s := range sites {
-				pr := idname.NewProvider(ptr.Val(s.GetId()), ptr.Val(s.GetName()))
+				pr := idname.NewProvider(ptr.Val(s.GetId()), ptr.Val(s.GetWebUrl()))
 				sbpc := inject.BackupProducerConfig{
 					LastBackupVersion:   bpc.LastBackupVersion,
 					Options:             bpc.Options,
```

[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/internal/m365/collection/drive/handler_utils.go:93]: Ah, good point.  Will do.

File: src/internal/m365/collection/drive/handler_utils.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/internal/m365/collection/drive/handler_utils.go:91]: Good point, I'll call it in with them.

Comment (meain) [reply src/internal/m365/collection/drive/handler_utils.go:91]: I had a chat. Looks like this is not a problem.

**Review by meain — APPROVED** [review]
LGTM

**Pull request merged**